### PR TITLE
Add socket data

### PR DESCRIPTION
**Pull request by C1200**
I saw that @y4my4my4m was unable to add this feature so I decided that I’d help

### CODE REVIEW

**Comment by y4my4my4m** [comment]
@C1200 makes sense, thank you very much for your help!
Would be great if we could include this. @darrachequesne

File: ui/src/components/Socket/SocketDetails.vue [hunk 1]
```diff
@@ -91,6 +91,12 @@
             <td></td>
           </tr>
 
+          <tr>
+            <td class="key-column">{{ $t("data") }}</td>
+            <td>{{ JSON.stringify(socket.data) }}</td>
```
Comment (darrachequesne) [inline ui/src/components/Socket/SocketDetails.vue:96]: Would it make sense to use something like this:

```html
<pre><code>{{ JSON.stringify(socket.data, null, 2) }}</code></pre>
```

to properly format the data?

File: lib/typed-events.ts [hunk 2]
```diff
@@ -26,6 +26,7 @@ export interface SerializedSocket {
   clientId: string;
   transport: string;
   nsp: string;
+  data: any;
```
Comment (darrachequesne) [inline lib/typed-events.ts:29]: The only downside is that the `data` property will not be sent to the client every time it is updated. Would it make sense to use a `Proxy` here? Or only refresh the data when the user visits the page of socket details?

File: lib/index.ts [hunk 3]
```diff
@@ -350,6 +350,7 @@ const serialize = (
     clientId,
     transport,
     nsp,
+    data: socket.data,
```
Comment (darrachequesne) [inline lib/index.ts:353]: We add an `_admin` attribute [here](https://github.com/socketio/socket.io-admin-ui/blob/9c8e7728a23db8f328848a9a05ae4ea17ab0f4ad/lib/index.ts#L306-L309), would it make sense to remove this attribute before sending the data to the user?

**Review by darrachequesne — COMMENTED** [review]
Thanks for your work on this! Please see my comments below.

**Comment by darrachequesne** [comment]
Merged as https://github.com/socketio/socket.io-admin-ui/commit/3773fe4b1cbf2206708e1f21ce65f430a522527f. Thanks :+1:

**Comment by y4my4my4m** [comment]
@darrachequesne 

image

https://admin.socket.io/

vercel deployment broke
https://github.com/socketio/socket.io-admin-ui/runs/6220530093?check_suite_focus=true

**Pull request closed without merging**